gapii/cc/vulkan_mid_execution.cpp

```diff
@@ -446,7 +455,7 @@ void VulkanSpy::serializeGPUBuffers(StateSerializer* serializer) {
         memory::Observation observation;
         observation.set_pool(deviceMemory->mData.pool_id());
         observation.set_base(bind.mmemoryOffset + offset);
-        serializer->sendData(&observation, true, stage.GetMappedMemory(),
+        serializer->sendData(&observation, true, stage->GetMappedMemory(),
```

Comment:
This also won't quite work. This will end up double-mapping memory. We should just map the memory on staging buffer creation, then we don't have to map it every time we need the contents.

Reply:
Isn't the `if (!bound_memory_)` in GetMappedMemory() forcing to map only once?

Reply:
My mistake, yes that's fine.